packages/viz/src/hooks/useBoxplotData.tsx

```diff
@@ -0,0 +1,52 @@
+import { useMemo } from "react";
+import { from, internal, table } from "arquero";
+import type ColumnTable from "arquero/dist/types/table/column-table";
+
+import { HvChartData } from "../types";
+import { HvChartCommonProps } from "../types/common";
+
+interface HvDataHookProps {
+  data: HvChartData;
+  groupBy: HvChartCommonProps["groupBy"];
+  measures: { [key: string]: string | undefined };
+}
+
+export const useBoxplotData = ({
+  data,
+  groupBy,
+  measures,
+}: HvDataHookProps) => {
+  const chartData = useMemo(() => {
+    let tableData: ColumnTable;
+    if (data instanceof internal.ColumnTable) {
+      tableData = data;
+    } else if (Array.isArray(data)) {
+      tableData = from(data);
+    } else {
+      tableData = table(data);
+    }
+
+    const groupByField = groupBy
```

Comment:
If we can only use one value for the `groupBy` prop, then we should update `HvBoxplotProps` and only accept a `string`.